ci-operator/step-registry/cucushift/installer/check/vsphere/zone/cucushift-installer-check-vsphere-zone-commands.sh

```diff
@@ -0,0 +1,156 @@
+   if [[ ${node_type} == "master" ]];then
+      readarray -t  node_distribution < <(yq-go r ${INSTALL_CONFIG} 'controlPlane.platform.vsphere.zones[*]')
+   else
+      readarray -t  node_distribution < <(yq-go r ${INSTALL_CONFIG} 'compute[*].platform.vsphere.zones[*]')
+   fi
+      mapfile -t  node_lists < <(oc get node --no-headers | grep ${node_type} | awk '{print $1}')
+    for node in "${node_lists[@]}"; do
+	    # shellcheck disable=SC2207
+            lable_info=($(oc get node ${node} --show-labels --no-headers | awk -F, '{print $(NF-1),$NF}'))
+            node_region=$(echo ${lable_info[0]} | awk -F= '{print $2}')
+            node_zone=$(echo ${lable_info[1]} | awk -F= '{print $2}')
+        for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
+                if [[ ${node_region} == "${zones_region_from_config[${i}]}" && ${node_zone} == "${zones_zone_from_config[${i}]}" ]]; then
+                    if [[ ${node_distribution[*]} =~ ${zones_name_from_config[${i}]} ]]; then
+                            echo "node ${node} located in correct failureDomain. zone name is ${zones_name_from_config[${i}]},region is ${zones_region_from_config[${i}]}, zone is ${zones_zone_from_config[${i}]} " 
+		    else   
+			    check_result=1
+			    echo "node ${node} located in incorrect failureDomain. zone name is ${zones_name_from_config[${i}]},region is ${zones_region_from_config[${i}]}, zone is ${zones_zone_from_config[${i}]} " 
+		    fi
+                fi
+        done
+
+    done
+ 
+return ${check_result}
+}
+
+function check_rhcos() {
+    check_result=0
+    infra_id=$(oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster)
+    # shellcheck disable=SC2207
+    vsphere_datacenters=($(govc ls /))
+    for vsphere_datacenter in "${vsphere_datacenters[@]}"; do
+        folder_name=$(govc ls ${vsphere_datacenter}/vm | grep ${infra_id} )
+	if [[ ${folder_name##*/} == "${infra_id}" ]]; then
+		echo " folder:${folder_name} under vcenter ${vsphere_datacenter} created successful.  "
+	else
+		check_result=1
+		echo " folder name is incorrect. "
+	fi
+    done	    
+
+# check all rhcos images uploaded into each failureDomain.
+        match_rhcos=0
+        for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
+	    rhcos_result=$(govc ls /${zones_datacenter_from_config[${i}]}/vm/${infra_id}/${infra_id}-rhcos-${zones_region_from_config[${i}]}-${zones_zone_from_config[${i}]} | wc -l)
+            
+	    if [[ ${rhcos_result} == 1 ]]; then
+		    match_rhcos=$(($match_rhcos+1))
+		    echo "rhcos image /${zones_datacenter_from_config[${i}]}/vm/${infra_id}/${infra_id}-rhcos-${zones_region_from_config[${i}]}-${zones_zone_from_config[${i}]} uploaded to specified failureDomain"
+            else 
+		    echo "rhcos image /${zones_datacenter_from_config[i]}/vm/${infra_id}/${infra_id}-rhcos-${zones_region_from_config[i]}-${zones_zone_from_config[i]} not found in specified failureDomain. please check"
+	            check_result=0
```

Comment:
we do not need this line line in this else branch, right?

Reply:
actually i want to set check_result=1 originally.